src/glmfit.jl

```diff
@@ -248,6 +248,30 @@ end
 
 deviance(m::AbstractGLM) = deviance(m.rr)
 
+function nulldeviance(m::AbstractGLM)
+    r   = m.rr
+    wts = r.wts
+    y   = r.y
+    d   = r.d
+
+    isempty(r.offset) ||
+        throw(ErrorException("models with offsets are currently not supported"))
+
+    dev = zero(eltype(y))
+    if length(wts) == length(y)
+        mu = mean(y, weights(wts))
+        @inbounds for i in eachindex(y, wts)
+            dev += wts[i] * devresid(d, y[i], mu)
+        end
+    else
```

Comment:
`wts` should either be empty of same length as `y`, right?

Reply:
Yeah, `isempty(wts)` would probably be clearer, and more robust since we would get an error anyway if lengths don't match. I just copied the pattern used in `loglikelihood` but I'll change all functions to use `isempty`.